pkg/domain/infra/abi/images.go

```diff
@@ -107,6 +108,11 @@ func (ir *ImageEngine) Prune(ctx context.Context, opts entities.ImagePruneOption
 		numPreviouslyRemovedImages = numRemovedImages
 	}
 
+	// Clean build cache if any
+	if err := volumes.CleanCacheMount(); err != nil {
```

Comment:
I'm a little uncertain about unconditionally blowing away on *any* `podman image prune` operation. For example today one can use `podman image prune --filter` to target specific images.

Combining things here...what if we e.g. added a label to buildah cache images, then one could target them with `podman image prune --filter`? As well as just a `podman image prune --build-cache` or so?

Reply:
Buildah's cache mount directory is different from intermediate build images which are used cached layers,  this part of code just ensures clearing cache directory which is created for `RUN --mount=type=cache` but I do agree that maybe adding a separate  `podman image prune --build-cache` is a good idea instead of silently cleaning it on every `podman image prune`.

I will amend the PR.

Reply:
Amended the PR added a dedicated `--build-cache` option.